dsf-tools/dsf-tools-documentation-generator/src/main/java/org/highmed/dsf/tools/generator/DocumentationGenerator.java

```diff
@@ -0,0 +1,222 @@
+				.setUrls(ClasspathHelper.forPackage(workingPackage))
+				.setScanners(Scanners.FieldsAnnotated, Scanners.SubTypes);
+		return new Reflections(configurationBuilder);
+	}
+
+	private List<String> getPluginProcessNames(Reflections reflections, String workingPackage)
+	{
+		List<Class<? extends ProcessPluginDefinition>> pluginDefinitionClasses = new ArrayList<>(
+				reflections.getSubTypesOf(ProcessPluginDefinition.class));
+
+		if (pluginDefinitionClasses.size() < 1)
+		{
+			logger.warn("No ProcessPluginDefinitions found in package {}", workingPackage);
+			return Collections.emptyList();
+		}
+
+		if (pluginDefinitionClasses.size() > 1)
+			logger.warn("Found {} ProcessPluginDefinitions ({}) in package {}, using only the first",
+					pluginDefinitionClasses.size(), pluginDefinitionClasses, workingPackage);
+
+		try
+		{
+			ProcessPluginDefinition processPluginDefinition = pluginDefinitionClasses.get(0).getConstructor()
+					.newInstance();
+
+			return processPluginDefinition.getBpmnFiles().map(this::getProcessName).filter(Optional::isPresent)
+					.map(Optional::get).collect(toList());
+		}
+		catch (Exception exception)
+		{
+			logger.warn(
+					"Could not read process names from package {} and ProcessPluginDefinition with name {}, reason is '{}'",
+					workingPackage, pluginDefinitionClasses.get(0).getName(), exception.getMessage());
+			return Collections.emptyList();
+		}
+	}
+
+	private Optional<String> getProcessName(String bpmnFile)
+	{
+		try
+		{
+			InputStream resource = getClass().getClassLoader().getResource(bpmnFile).openStream();
+			return Bpmn.readModelFromStream(resource).getModelElementsByType(Process.class).stream()
+					.map(BaseElement::getId).findFirst();
+		}
+		catch (Exception exception)
+		{
+			logger.warn("Could not read process name from resource file {}, reason is '{}'", bpmnFile,
+					exception.getMessage());
+			return Optional.empty();
+		}
+	}
+
+	private List<Field> getFields(Reflections reflections)
+	{
+		List<Field> fields = new ArrayList<>(reflections.getFieldsAnnotatedWith(Documentation.class));
+		Collections.reverse(fields);
+		return fields;
+	}
+
+	private void writeFields(List<Field> fields, List<String> pluginProcessNames, String filename,
+			String workingPackage)
+	{
+		try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename)))
+		{
+			fields.stream().map(field -> createDocumentation(field, pluginProcessNames))
+					.forEach(d -> write(writer, filename, d));
+		}
+		catch (Exception exception)
+		{
+			logger.warn("Could not generate documentation for package {}, reason is '{}'", workingPackage,
+					exception.getMessage());
+		}
+	}
+
+	private String createDocumentation(Field field, List<String> pluginProcessNames)
+	{
+		Documentation documentation = field.getAnnotation(Documentation.class);
+		Value value = field.getAnnotation(Value.class);
+
+		String[] valueSplit = getValueDefaultArray(value.value());
+
+		String initialProperty = valueSplit[0];
```

Comment:
I have not looked closely: Is valueSplit always != null with length >= 1 ?

Reply:
resolved with 6a6eb6f